nemo/core/classes/modelPT.py

```diff
@@ -1246,3 +1245,22 @@ def set_eff_save(use_eff_save: bool):
     def use_eff_save() -> bool:
         global _MODEL_EFF_SAVE
         return _MODEL_EFF_SAVE
+
+    def load_weights_from_checkpoint(self, model_path, strict=True):
```

Comment:
Rename this function - checkpoint implies `PTL .ckpt` but this function takes both `ckpt` as well as `.nemo`. Also, ckpt can be confused for Pytorch .pt - which is not supported.

This method has a fatal flaw - it tries to restore model using `self` - but current context of `self` is ModelPT. So while the method exists in ModelPT, trying `ModelPT.load_weights_from_checkpoint(...)` will fail since `self` doesnt resolve to the actual class that needs to be loaded up. I will send a PR to generalize this (at least for .nemo files). However this will fail for PTL checkpoints.

I think we should not allow PTL checkpoints at all in such a function, only .nemo files. For PTL files, experiment manager has its `resume*` flags and only those should be the supported path to avoid duplicating complexity. @okuchaiev does that sound fine? We mostly release pretrained .nemo files only anyway.

Reply:
Add docstrings to this function.

Reply:
+ I have tested these functions with both nemo and PTL checkpoints. They work fine. Python has dynamic binding so when we call that methosd it calls the right method based on the real type of the object.

+ Current exp_manager does not support this feature currently. This feature is different from what we have currently. The one we have is to continue training which loads both the weights and optimizer state, etc. But this feature just loads the weights which is used for fine-tuning. Fine-tuning is different from continuing previous runs as we should not just load the weights and nothing else. I think currently the exp_manager can not be used for fine-tuning without adding some lines to the speech_to_text_bpe.py. Do you currently fine-tune with speech_to_text.py and exp_manager?

Reply:
+ Currently I think PTL handles resume_from_checkpoint with the same approach. The only difference is that they load the whole state including config and optimizer state while this feature loads just weights.

+ We can add this feature of just loading the weights to the current load models/checkpoints method. It is fine. Then we can call them. But that is not the main goal of this PR. This PR is to enable all the examples in the NeMo to support fine-tuning without the need of the user to add extra lines to the examples. It is not just for text_to_speech. It is a general feature.  Currently all NeMo users who do fine-tuning need to add extra lines (it they have enough knowledge) to the examples to perform fine-tuning.

Reply:
+ added @ericharper as we are making changes to the core.

Reply:
.nemo files are meant to be used for finetuning - you are not supposed to load up a model seperately, then load state dict for this model from yet another .nemo / PTL checkpoint. First - you should not even be using .ptl checkpoint for finetuning, since as you mentioned it contains the optimizer state which is incorrect for finetuning!

If this is meant for just finetuning, then this function should drop support for PTL entirely since it is incorrect.

This PR adds yet another approach to fine tune - which is by loading one model via config, restoring some/all of its weights from another checkpoint (this can be .nemo / ptl .ckpt but not pt ckpt), and then continue training. I dont see how this is a good practice, but there are cases for it I guess. @okuchaiev needs to decide if we want to support this. Only exp manager should work on .ptl checkpoints.

To note, we will now have 4 different ways of handling checkpoints - and that doesnt make sense to me.
1) restore_from for .nemo  (for finetuning or eval)
2) from pretrained for .nemo (for finetuning or eval)
3) load_state_dict - for PT / PTL checkpoints  (for finetuning or eval)
4) and now load_state_from_checkpoints which is for both .nemo and ptl checkpoints (but not PT checkpoints) - (for finetuning or eval).

Reply:
This PR also doesnt change the fact that users will still need to add these lines to the example scripts - so other than hiding 3-4 lines inside ModelPT - it has no major benefit. All of the example scripts will need to update to use this function anyway.